### PR TITLE
pkp/pkp-lib#2392 migrate removed to the existing fields on the about …

### PR DESCRIPTION
**Pull request by bozana**
…page

s. https://github.com/pkp/pkp-lib/issues/2392

### CODE REVIEW

File: classes/install/Upgrade.inc.php [hunk 1]
```diff
@@ -1930,6 +1930,277 @@ function removeCancelledReviewAssignments() {
...
+	 * @return boolean
+	 */
+	function concatenateIntoAbout() {
+		$journalDao = DAORegistry::getDAO('JournalDAO');
+		$journalSettingsDao = DAORegistry::getDAO('JournalSettingsDAO');
+		$journals = $journalDao->getAll();
+		while ($journal = $journals->next()) {
+			$supportedFormLocales = $journal->getSupportedFormLocales();
+			$focusAndScope = $journal->getSetting('focusScopeDesc');
+			$focusAndScope['localKey'] = 'about.focusAndScope';
```
Comment (asmecher) [inline classes/install/Upgrade.inc.php]: I think `localKey` should be `localeKey` throughout, no?

File: classes/install/Upgrade.inc.php [hunk 2]
```diff
@@ -1930,6 +1930,277 @@ function removeCancelledReviewAssignments() {
...
+						if (!empty($customItem['title']) && !empty($customItem['content'])) {
+							$aboutJournal[$locale] .= '
+								<div id="custom-'.$index.'">
+								<h3>'.$customItem['title'].'</h3>
+								<p>'.nl2br($customItem['content']).'</p>
+								</div>';
+						}
+					}
+				}
+				// finaly, concatenate the other settings
```
Comment (asmecher) [inline classes/install/Upgrade.inc.php]: *finally

File: classes/install/Upgrade.inc.php [hunk 3]
```diff
@@ -1930,6 +1930,277 @@ function removeCancelledReviewAssignments() {
...
+		}
+		return true;
+	}
+
+	/**
+	 * For 2.4.x - 3.1.0 upgrade: concatenate editorialTeam and displayMembership page to new journal setting 'masthead'.
+	 * @return boolean
+	 */
+	function concatenateIntoMasthead() {
+		$roleIdEditor = 0x00000100;
```
Comment (asmecher) [inline classes/install/Upgrade.inc.php]: I'd suggest putting the OJS 2.x constant names in comments -- it'll help when grepping and explain where the magic numbers come from.

Comment (asmecher) [inline classes/install/Upgrade.inc.php]: (I still prefer `define` statements for these, but it's not a big enough deal to warrant changing)

Comment (bozana) [inline classes/install/Upgrade.inc.php]: They are defined in a function above, so I thought it is more secure like this, depending on where and when they are needed, which function is when executed... to have them defined in the appropriate function... OK?

Comment (asmecher) [inline classes/install/Upgrade.inc.php]: Ah, I see. Hmm, maybe we should define them once at the top of the file, something like `OLD_ROLE_ID_...`? That way grepping will work, the constants will just be defined once, and magic numbers get documented.

Comment (bozana) [inline classes/install/Upgrade.inc.php]: OK, I can do that :-) THANKS!!!

**Pull request merged**